parlai/agents/transformer/modules/decoder.py

```diff
@@ -20,12 +22,160 @@
     MultiHeadAttention,
     TransformerFFN,
 )
+from parlai.agents.transformer.modules.interfaces import ComponentSpec, TComponent
 from parlai.core.opt import Opt
 from parlai.utils.misc import warn_once
 from parlai.utils.torch import PipelineHelper
 
 
-class TransformerDecoder(nn.Module):
+class TransformerDecoderLayer(nn.Module, TComponent):
```

Comment:
Now that we're actually _calling_ an attribute of this class later in the file instead of just referencing it as a type annotation we need to move it up.